connector/mqtt/latency-pub-test-prometheus/app/LatencyStore.js

```diff
@@ -0,0 +1,111 @@
+const ss = require('simple-statistics');
+const { logger } = require('@dojot/dojot-module-logger');
+
+const TAG = { filename: 'LatencyStore' };
+
+/** Class to store Latencys and calculate statistics * */
+class LatencyStore {
+  /**
+     *
+     */
+  constructor() {
+    logger.debug('Init LatencyStore...', TAG);
+
+    this.latencies = [];
+    this.max = null;
+    this.min = null;
+  }
+
+  /**
+     * Retrieve array of latencies
+     *
+     * @return {Array} All latencies
+     */
+  getLatencies() {
+    return this.latencies;
+  }
+
+  /**
+     * Clean all store info
+     */
+  cleanLatencies() {
+    logger.debug(`Clean all Latencies, the Latencies before clean: ${this.getLatencies()}`, TAG);
+
+    this.latencies = [];
+    this.max = null;
+    this.min = null;
+  }
+
+  /**
+     * Add a new latency to store
+     *
+     * @param {number} latency in ms
+     */
+  addLatency(latency) {
+    this.latencies.push(latency);
+
+    this.updateMax(latency);
+    this.updateMin(latency);
+  }
+
+  /**
+     * Update the attrs of classe min if param is less then current
+     *
+     * @param {number} Latency
+     */
+  updateMin(latency) {
+    this.min = this.min === null || this.min > latency ? latency : this.min;
+  }
+
+  /**
+     * Update the attrs of classe max if param is more then current
+     *
+     * @param {number} Latency
+     */
+  updateMax(latency) {
+    this.max = this.max === null || this.max < latency ? latency : this.max;
+  }
+
+  /**
+     * Get maximum value for all latencies stores
+     * @return {number} Maximum for all latencies stores
+     */
+  getMax() {
+    return this.max;
+  }
+
+  /**
+     * Get mininum value for all latencies stores
+     * @return {number} Mininum for all latencies stores
+     */
+  getMin() {
+    return this.min;
+  }
+
+  /**
+    * Get average value for all latencies stores
+    * @return {number} Average for all latencies stores
+    */
+  getAvg() {
+    return this.getLatencies().length > 0 ? ss.mean(this.getLatencies()) : null;
+  }
+
+  /**
+    * Get median value for all latencies stores
+    * @return {number} Median for all latencies stores
+    */
+  getMedian() {
+    return this.getLatencies().length > 0 ? ss.median(this.getLatencies()) : null;
+  }
+
+  /**
+    * Get Standard Deviation value for all latencies stores
+    * @return {number} Standard Deviation for all latencies stores
+    */
+  getStandardDeviation() {
+    return this.getLatencies().length > 0 ? ss.standardDeviation(this.getLatencies()) : null;
+  }
+}
+
+const latencyStore = new LatencyStore();
```

Comment:
why instantiating a new class then export it?
If we would like to use only one instance in the whole application, it can be initialized inside the main, or can it?

Reply:
Why should I instance on main?

Reply:
It just a question, because everyone who import/requires this class, wants to instantiate it not to requires an instantiated class

Reply:
So, I would like to use a singleton, there are other ways to do this, but I don't know exactly why not use it when exporting an instance?